cborg.js

```diff
@@ -2,6 +2,13 @@ import { encode } from './lib/encode.js'
 import { decode, decodeFirst, Tokeniser, tokensToObject } from './lib/decode.js'
 import { Token, Type } from './lib/token.js'
 
+// is this needed for the json module and other independ encoders
+export { encodeCustom } from './lib/encode.js'
+export { encodeErrPrefix, decodeErrPrefix } from './lib/common.js'
+export { asU8A, fromString, decodeCodePointsArray } from './lib/byte-utils.js'
+export { quickEncodeToken } from './lib/jump.js'
+export { makeCborEncoders, objectToTokens } from './lib/encode.js'
```

Comment:
this can't happen here, for two reasons:

1. Users who don't need JSON support (a large portion of them), shouldn't be forced to get these, they should be entirely absent from the dependency tree and not add weight to a bundle if you don't use it. I know this seems to be something you're trying to solve for because you're using JSON, but this library is mainly about CBOR. So we need to find an alternative to doing this. If it means re-exporting a top-level export just for JSON that includes everything needed then that's fine.
2. Anything exported on the top-level becomes part of the API contract and there's far too many things here that I don't want to be on the hook for supporting. Like `asU8A`, this is not something that I want to expose for general use, any time I want to make an internal change that involves these functions I now need to be thinking about whether it's a breaking change or not. Internal stuff _must_ remain internal and these all look internal to me.

Reply:
Thx,

I will run through it and try to move things around. 
I would argue if they are used at the top-level namespace implementation the export does not reflect the bundle size.

Reply:
Hi,

I did the move around --- ending up renaming lib -> cborg (which is more a hygiene thing)
I tried hard to move 
- encodeCustom
but due to that's the real entry point for a configured encoder there is no simple way to
no export this from the main package.
This thing is a bit bigger than expected due to the de/encodeErrPrefix, which is now configurable.

If you are happy with that -> I will remove the comments that I left for now.

Pls do review and give some feedback. @rvagg 

Due to my renaming of the directory, we should not have this PR lying around for too long.